sdk/communication/AzureCommunicationCommon/Source/Identifiers.swift

```diff
@@ -114,31 +138,65 @@ import Foundation
         rawId: String? = nil,
         cloudEnvironment: CommunicationCloudEnvironment = .Public
     ) {
-        self.rawId = rawId
         self.userId = userId
         self.isAnonymous = isAnonymous
         self.cloudEnviroment = cloudEnvironment
+
+        if let rawId = rawId {
+            self.rawId = rawId
+        } else {
+            if isAnonymous {
+                self.rawId = "8:teamsvisitor:" + userId
+            } else {
+                switch cloudEnvironment {
+                case .Dod:
+                    self.rawId = "8:dod:" + userId
+                case .Gcch:
+                    self.rawId = "8:gcch:" + userId
+                case .Public:
+                    self.rawId = "8:orgid:" + userId
+                default:
+                    self.rawId = "8:orgid:" + userId
+                }
+            }
+        }
     }
 
+    /**
+     Creates a MicrosoftTeamsUserIdentifier object. cloudEnvironment is defaulted to Public cloud.
+     - Parameter userId: Id of the Microsoft Teams user. If the user isn't anonymous,
+                            the id is the AAD object id of the user.
+     - Parameter isAnonymous: Set this to true if the user is anonymous:
+                                for example when joining a meeting with a share link.
+     */
     public init(userId: String, isAnonymous: Bool) {
         self.cloudEnviroment = .Public
-        self.rawId = nil
         self.userId = userId
         self.isAnonymous = isAnonymous
+        self.rawId = isAnonymous ? "8:teamsvisitor:" + userId : "8:orgid:" + userId
```

Comment:
I suggest we keep the logic for constructing the `rawId` at a single place (right now it's partially duplicated in two constructors). 
plus, not sure if it's idiomatic for swift, but I'd prefer to have the logic outside of the constructor...or at least extracted to a method.

Reply:
Unfortunately in swift because `rawId` is required. I cannot have a helper function called in the constructor until all the  properties of the class have been assigned. If we wanted to move it into a helper function it would need to be a static method and I don't think that is the right approach

Reply:
I see. Then perhaps the convenience init could potentially make the code simpler.